server/src/main/java/org/elasticsearch/index/query/AbstractQueryBuilder.java

```diff
@@ -194,27 +195,31 @@ public final int hashCode() {
     protected abstract int doHashCode();
 
     /**
-     * This helper method checks if the object passed in is a string, if so it
-     * converts it to a {@link BytesRef}.
+     * This helper method checks if the object passed in is a string or {@link CharBuffer},
+     * if so it converts it to a {@link BytesRef}.
      * @param obj the input object
      * @return the same input object or a {@link BytesRef} representation if input was of type string
      */
-    static Object convertToBytesRefIfString(Object obj) {
```

Comment:
I would go with a simpler name since you are renaming anwyays: `convertToBytesRef`. I don't think the "if" really adds anything that can't/isn't explained in the javadoc.

Reply:
How about `maybeConvertToBytesRef`? Because it doesn't always convert the object to a BytesRef

Reply:
Well, I think methods like this are trappy to begin with. We should have types, not be converting object to object. But that's outside the scope of this PR. Using `maybe` is fine I guess.

Reply:
Yeah, I definitely don't like this method at all, using `Object` is definitely not something I want to do, but I agree it's out of scope for this PR.

Reply:
++